packages/react-reconciler/src/ReactFiberCommitWork.new.js

```diff
@@ -1964,6 +1964,10 @@ function commitMutationEffects_begin(
     if (deletions !== null) {
       for (let i = 0; i < deletions.length; i++) {
         const childToDelete = deletions[i];
+        // When deleting a host node, we follow the return path to find the
+        // neartest mounted host parent. Which means the return pointer of
+        // the deleted tree must be correct.
```

Comment:
Instead of doing that on-demand, we could do that eagerly here and then pass in the parent.

It's likely to be at least one DOM node deleted. If it's more than one, then it's unnecessary to backtrack more than once. We meant to do that optimization for that reason at some point anyway.

Reply:
We don't use recursion for this function yet (was reverted because of slow down in open source build) so it'd have to be a virtual stack.

So I'll go with option 1 (eagerly find the current parent) for now and leave a TODO for when this recursive traversal is converted to use the JS stack.